test/realm-fuzzer/fuzz_object.cpp

```diff
@@ -0,0 +1,541 @@
+                    else {
+                        log << "obj.get_linklist(" << col << ")->add(" << target_key << ");\n";
+                        links.add(target_key);
+                    }
+                }
+            }
+            else if (type == type_Timestamp) {
+                std::pair<int64_t, int32_t> values = get_timestamp_values(s);
+                Timestamp value{values.first, values.second};
+                log << "obj.set(" << col << ", " << value << ");\n";
+                obj.set(col, value);
+            }
+        }
+        log << "}\n";
+    }
+    else {
+        log << "table " << table_key << " has size = " << t->size()
+            << " and get_column_keys size = " << all_col_keys.size() << "\n";
+    }
+}
+
+void FuzzObject::remove_obj(Group& group, FuzzLog& log, State& s)
+{
+    log << "FuzzObject::remove_obj();\n";
+    TableKey table_key = group.get_table_keys()[get_next_token(s) % group.size()];
+    TableRef t = group.get_table(table_key);
+    if (t->size() > 0) {
+        ObjKey key = t->get_object(get_next_token(s) % t->size()).get_key();
+        log << "group.get_table(" << table_key << ")->remove_object(" << key << ");\n";
+        t->remove_object(key);
+    }
+}
+
+void FuzzObject::remove_recursive(Group& group, FuzzLog& log, State& s)
+{
+    log << "FuzzObject::remove_recursive();\n";
+    TableKey table_key = group.get_table_keys()[get_next_token(s) % group.size()];
+    TableRef t = group.get_table(table_key);
+    if (t->size() > 0) {
+        ObjKey key = t->get_object(get_next_token(s) % t->size()).get_key();
+        log << "group.get_table(" << table_key << ")->remove_object_recursive(" << key << ");\n";
+        t->remove_object_recursive(key);
+    }
+}
+
+void FuzzObject::enumerate_column(Group& group, FuzzLog& log, State& s)
+{
+    log << "FuzzObject::enumerate_column();\n";
+    TableKey table_key = group.get_table_keys()[get_next_token(s) % group.size()];
+    TableRef t = group.get_table(table_key);
+    auto all_col_keys = t->get_column_keys();
+    if (!all_col_keys.empty()) {
+        size_t ndx = get_next_token(s) % all_col_keys.size();
+        ColKey col = all_col_keys[ndx];
+        log << "group.get_table(" << table_key << ")->enumerate_string_column(" << col << ");\n";
+        group.get_table(table_key)->enumerate_string_column(col);
+    }
+}
+
+void FuzzObject::get_all_column_names(Group& group, FuzzLog& log)
+{
+    log << "FuzzObject::get_all_column_names();\n";
+    for (auto table_key : group.get_table_keys()) {
+        TableRef t = group.get_table(table_key);
+        auto all_col_keys = t->get_column_keys();
+        for (auto col : all_col_keys) {
+            StringData col_name = t->get_column_name(col);
+            static_cast<void>(col_name);
+        }
+    }
+}
+
+void FuzzObject::commit(SharedRealm shared_realm, FuzzLog& log)
+{
+    log << "FuzzObject::commit();\n";
+    log << "FuzzObject::commit() - shared_realm->is_in_transaction();\n";
```

Comment:
I am not sure if I understood the comment, the point of the log is to help us to debug the issue. The crash report will serve as input to reproduce the issue.

Reply:
The crash report is a good first step to reproduce the issue, but it is not human readable. In the previous AFL based fuzz tests, the purpose of the log file was to be able to copy and paste the output into a core unit test so that we could debug the issue further. The logs generated by this code are not compilable code without being modified, this would be fine for a small amount of code, but in my experience, the logs generated are many hundreds of lines long which would be very tedious to transform by hand into a unit test. However, this is more of a nice to have feature so it is not blocking for me.

Reply:
I can try to reimplement this feature, but I would like to get this PR into master first.